File: Proxy/proxy01.js
// Objeto Real
class ServicoReal{
    executar(){
        console.log("Objeto Real - Executando o serviço real.");
    }
}

// Proxy (Procurador)
class ProxyServicoReal{
    constructor(){
        this.servicoReal = new ServicoReal();
    }

    executar(){
        console.log("Proxy - Antes da Execução do Serviço");
        this.servicoReal.executar();
        console.log("Proxy - Depois da Execução do Serviço");
    }
};

// Uso do Proxy:
const proxy = new ProxyServicoReal();
proxy.executar();

File: Proxy/proxy01.py
# Objeto Real
class ServicoReal:
    def executar(self):
        print("Objeto Real - Executando o Serviço Real")

# Proxy (Procurador)
class ProxyServicoReal:
    def __init__(self):
        self.servicoReal = ServicoReal();

    def executar(self):
        print("Proxy - Antes da Execução do Serviço")
        self.servicoReal.executar()
        print("Proxy - Depois da Execução do Serviço")

# Uso do Proxy:
proxy = ProxyServicoReal()
proxy.executar()